apps/emqx_web_hook/src/emqx_web_hook_actions.erl

```diff
@@ -385,4 +386,10 @@ parse_host(Host) ->
                 {ok, _} -> {inet6, Host};
                 {error, _} -> {inet, Host}
             end
+    end.
+
+test_http_connect(Url) ->
+    case emqx_rule_utils:http_connectivity(Url) of
+        ok -> ok;
+        {error, _} -> error({error, http_connect_failure})
```

Comment:
```suggestion
test_http_connect(Conf) ->
  try emqx_rule_utils:http_connectivity(maps:get(<<"url">>, Conf)) of
     ok -> true;
     {error, Reason} ->
       ?LOG(error, "check http_connectivity failed: ~p", [Url]),
       false
  catch
      Err:Reason:ST ->
         ?LOG(error, "check http_connectivity failed: ~p, ~0p", [Url, {Err,Reason,ST}]),
         false
  end
```

Reply:
And then use this function in both `on_resource_create /2` and `on_get_resource_status /2`.